coderd/webpush/webpush.go

```diff
@@ -0,0 +1,236 @@
+package webpush
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"io"
+	"net/http"
+	"slices"
+	"sync"
+
+	"github.com/SherClockHolmes/webpush-go"
+	"github.com/google/uuid"
+	"golang.org/x/sync/errgroup"
+	"golang.org/x/xerrors"
+
+	"cdr.dev/slog"
+	"github.com/coder/coder/v2/coderd/database"
+	"github.com/coder/coder/v2/coderd/database/dbauthz"
+	"github.com/coder/coder/v2/codersdk"
+)
+
+// Dispatcher is an interface that can be used to dispatch
```

Comment:
Hmm.. you are referring to web push notifications. For clarity, I would refer to web push message wherever possible.